cloud-aws-common/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/common/util/AwsStackNameCommonUtil.java

```diff
@@ -0,0 +1,37 @@
+package com.sequenceiq.cloudbreak.cloud.aws.common.util;
+
+import static org.slf4j.LoggerFactory.getLogger;
+
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+
+import com.sequenceiq.cloudbreak.cloud.context.AuthenticatedContext;
+
+@Component
+public class AwsStackNameCommonUtil {
+
+    private static final Logger LOGGER = getLogger(AwsStackNameCommonUtil.class);
+
+    private static final String INSTANCE_NAME_PATTERN = "%s-%s%d";
+
+    @Value("${cb.max.aws.resource.name.length:}")
+    private int maxResourceNameLength;
+
+    public String getInstanceName(AuthenticatedContext ac, String groupName, Long privateId) {
+        if (StringUtils.isBlank(groupName)) {
+            throw new IllegalArgumentException("Group name cannot be empty, instance name cannot be generated");
+        }
+        int lengthOfPostFix = privateId.intValue() + groupName.length();
+        int maxLengthOfName = maxResourceNameLength - lengthOfPostFix;
+        String stackName = ac.getCloudContext().getName();
+        if (stackName.length() > maxLengthOfName) {
+            stackName = stackName.substring(0, maxLengthOfName - 1);
```

Comment:
Sanity check: Is the `- 1` intentional? The end index of `substring()` is exclusive, so this will result in a string with a length of at most `maxLengthOfName - 1` characters (so one shorter than the maximum),

Reply:
yes, I'm going to add this check to the unit test as well